R/utils.R

```diff
@@ -30,4 +30,53 @@ is_atomic_list <- function(ll, .p) {
 
 assertthat::on_failure(is_atomic_list) <- function(call, env) {
   "Element of the list are not all of the same class"
-}
\ No newline at end of file
+}
+
+
+#'Build a `VolSurface` from an example date set
+#'
+#'This creates a [`VolSurface`][`VolSurface`] object from the example data set
+#' `volsurface.csv`.
+#'
+#' @return a `VolSurface` object using data from `volsurface.csv`
+#' @examples build_vol_surface()
+#' @export
+#' @family build object helpers
+
+build_vol_surface <- function(){
+##1.651555326
+
+  filepath <- system.file("extdata", "volsurface.csv", package = "fmbasics")
+  vols <- utils::read.csv(file = filepath, header = T )
+  spot <- 97.62
+  reference_date <- as.Date("2019-04-26")
+  tenors <- lubridate::ymd(vols$Date)
+  strikes <- seq(from = 0.05, to = 2.55, by= 0.05)*spot
```

Comment:
Why do you need to hard code this - isn't this in `volsurface.csv"?

Reply:
The data in "volsurface.csv" does not include this information (ref date, tenors, strike) explicitly. It only contains the tenor dates in the form yyyymmdd, the strikes are given via moneyness: X5., X10., X15., ...., X250., where the digits refer to (K/S)*100. This is an example of a vol surface extracted from Murex.

Reply:
![image](https://user-images.githubusercontent.com/39390900/62014308-96aa9e00-b1e2-11e9-9067-d56557263c1d.png)

 This is the format of  "vols" data sourced from the exdata (volsurface.csv). I now use the first line (colnames) containing the ratio information to derive the moneyness and therefore the strikes.